compiler/aarch64/codegen/OMRMachine.cpp

```diff
@@ -19,11 +19,457 @@
  * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
  *******************************************************************************/
 
+#include "aarch64/codegen/OMRMachine.hpp"
+
+#include <stdint.h> // for uint16_t, int32_t, etc
+
 #include "codegen/CodeGenerator.hpp"
 #include "codegen/Machine.hpp"
+#include "codegen/Machine_inlines.hpp"
+#include "codegen/RealRegister.hpp" // for RealRegister, etc
+#include "infra/Assert.hpp" // for TR_ASSERT
 
 OMR::ARM64::Machine::Machine(TR::CodeGenerator *cg) :
-      OMR::Machine(cg)
+      OMR::Machine(NUM_ARM64_GPR, NUM_ARM64_FPR)
+   {
+   _registerFile = (TR::RealRegister **)cg->trMemory()->allocateMemory(sizeof(TR::RealRegister *)*TR::RealRegister::NumRegisters, heapAlloc);
+   self()->initialiseRegisterFile();
+   }
+
+TR::RealRegister *OMR::ARM64::Machine::findBestFreeRegister(TR_RegisterKinds rk,
+                                                            bool excludeGPR0,
+                                                            bool considerUnlatched)
    {
+   int first;
+   int last;
+
+   switch(rk)
+      {
+      case TR_GPR:
+         first = TR::RealRegister::FirstGPR;
+         if (excludeGPR0)
+            first++;
+         last  = TR::RealRegister::LastAssignableGPR;
+         break;
+      case TR_FPR:
+         first = TR::RealRegister::FirstFPR;
+         last  = TR::RealRegister::LastFPR;
+         break;
+   }
+
+   uint32_t bestWeightSoFar = 0xffffffff;
```

Comment:
At some point we need to decide whether register weights make sense for ARM or would have any benefit.  We can't necessarily predict how some unforeseen linkages or code paths may want to influence the register selection choices, so I'm fine with leaving the basic infrastructure in place.